src/os/track.js

```diff
@@ -609,25 +605,71 @@ os.track.truncate = function(track, size) {
 
   if (geometry.getType() === ol.geom.GeometryType.MULTI_LINE_STRING) {
     // merge the split line so coordinates can be truncated to the correct size
-    geometry.toLonLat();
-    geometry = os.geo.mergeLineGeometry(geometry);
-    geometry.osTransform();
+    os.track.mergeSplitGeometry_(geometry);
   }
 
-  var flatCoordinates = geometry.flatCoordinates;
   var stride = geometry.stride;
   var numCoords = size * stride;
+  os.track.prunePointsAt_(track, geometry, numCoords);
+};
+
+/**
+ * Merge a split line geometry.
+ * @param {!os.track.TrackLike} geometry the geometry to merge
+ */
+os.track.mergeSplitGeometry_ = function(geometry) {
+  geometry.toLonLat();
+  geometry = os.geo.mergeLineGeometry(geometry);
+  geometry.osTransform();
+};
 
+/**
+ * Remove old track points from a track. Keeps the most recent points.
+ *
+ * @param {!ol.Feature} track The track.
+ * @param {number} maximumAge The maximum age that a track point can get to before it is considered for removal, in milliseconds.
+ *
+ * @suppress {accessControls} To allow direct access to feature metadata and line coordinates.
+ */
+os.track.truncateByAge = function(track, maximumAge) {
+  // ensure the age-out is >= 0
+  maximumAge = Math.max(0, maximumAge);
+
+  // add point(s) to the original geometry, in case the track was interpolated
+  var geometry = /** @type {!(os.track.TrackLike)} */ (track.values_[os.interpolate.ORIGINAL_GEOM_FIELD] ||
+      track.getGeometry());
+
+  if (geometry.getType() === ol.geom.GeometryType.MULTI_LINE_STRING) {
+    // merge the split line so coordinates can be counted and truncated cleanly
+    os.track.mergeSplitGeometry_(geometry);
+  }
+  var coords = geometry.getCoordinates();
+  var timeNow = Date.now();
+  var pointsToKeep = 0;
+  if (coords) {
+    coords.reverse().forEach(function(c) {
```

Comment:
`os.track.getTimeIndex` will get the closest index to a specified timestamp using binary search, so it may be more performant to:

- Get the closest index to the age-off timestamp.
- Splice the flat coordinates array from the index to the end.
- Do the same for metadata.

This would avoid the `getCoordinates` call which reconstitutes the flat coordinates (unnecessary GC of all those arrays), and reversing/iterating over the entire array.